fermipy/lightcurve.py

```diff
@@ -28,6 +31,181 @@
+        elif niter == 2:
+            gta.logger.info('Fit Failed with User Supplied List of '
+                             'Free/Fixed Sources.....Lets try '
+                             'fixing TS<4 sources')
+            gta.free_sources_by_name(free_sources, False)
+            gta.free_sources_by_name(free_sources, pars='norm')
+            gta.free_sources(minmax_ts=[0, 4], free=False, exclude=[name])
+        elif niter == 3:
+            gta.logger.info('Fit Failed with User Supplied List of '
+                             'Free/Fixed Sources.....Lets try '
+                             'fixing TS<9 sources')
+            gta.free_sources_by_name(free_sources, False)
+            gta.free_sources_by_name(free_sources, pars='norm')
+            gta.free_sources(minmax_ts=[0, 9], free=False, exclude=[name])
+        elif niter == 4:
+            gta.logger.info('Fit still did not converge, lets try fixing the '
+                             'sources up to 1dg out from ROI')
+            gta.free_sources_by_name(free_sources, False)
+            for s in free_sources:
+                src = gta.roi.get_source_by_name(s)
+                if src['offset'] < 1.0:
+                    gta.free_source(s, pars='norm')
+            gta.free_sources(minmax_ts=[0, 9], free=False, exclude=[name])
+        else:
+            gta.logger.error('Fit still didnt converge.....please examine this data '
+                              'point, setting output to 0')
+            break
+
+        fit_results = gta.fit()
+        if fit_results['fit_success'] is True:
+            break
+
+    return fit_results
+
+def _process_lc_bin(itime, name, config, basedir, workdir, diff_sources, const_spectrum, **kwargs):
+    i, time = itime
+
+    config = copy.deepcopy(config)
+    config['selection']['tmin'] = time[0]
+    config['selection']['tmax'] = time[1]
+    config['ltcube']['use_local_ltcube'] = kwargs['use_local_ltcube']
+    config['model']['diffuse_dir'] = [workdir]
+
+    if config['components'] is None:
+        config['components'] = []
+
+    # create output directories labeled in MET vals
+    outdir = basedir + 'lightcurve_%.0f_%.0f' % (time[0], time[1])
+    config['fileio']['outdir'] = os.path.join(workdir, outdir)
+    utils.mkdir(config['fileio']['outdir'])
+
+    yaml.dump(utils.tolist(config),
+              open(os.path.join(config['fileio']['outdir'],
+                                'config.yaml'), 'w'))
+
+    xmlfile = os.path.join(config['fileio']['outdir'], 'base.xml')
+
+    # Make a copy of the source maps. TODO: Implement a
+    # correction to account for the difference in exposure for
+    # each time bin.
+    #     for c in self.components:
+    #        shutil.copy(c._files['srcmap'],config['fileio']['outdir'])
+
+    try:
+        from fermipy.gtanalysis import GTAnalysis
+        gta = GTAnalysis(config)
+        gta.logger.info('Fitting time range %i %i' % (time[0], time[1]))
+        for j, c in enumerate(gta.components):
```

Comment:
Is there a reason you moved this part of the configuration after the instantiation of the `GTAnalysis` object?  It's generally not recommended to change the configuration state after the analysis object is created.

Reply:
The reason here was I was updating the old code which looped over self.components.

Reply:
Ok i see.  I think we can move this part of the logic back into `_make_lc` since this part of the configuration is common to all lightcurve bins.  Working on this now.